v2/internal/controllers/crd_keyvault_test.go

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package controllers_test
+
+import (
+	"testing"
+
+	keyvault "github.com/Azure/azure-service-operator/v2/api/keyvault/v1beta20210401preview"
+	resources "github.com/Azure/azure-service-operator/v2/api/resources/v1beta20200601"
+	"github.com/Azure/azure-service-operator/v2/internal/testcommon"
+	"github.com/Azure/go-autorest/autorest/to"
+)
+
+func Test_KeyVault_Vault_CRUD(t *testing.T) {
+
+	t.Parallel()
+
+	tc := globalTestContext.ForTest(t)
+
+	rg := tc.CreateTestResourceGroupAndWait()
+	defer tc.DeleteResourcesAndWait(rg)
+
+	vault := newVault(tc, rg)
+
+	tc.CreateResourceAndWait(vault)
+	tc.DeleteResourceAndWait(vault)
+
+}
+
+func newVault(tc *testcommon.KubePerTestContext, rg *resources.ResourceGroup) *keyvault.Vault {
+
+	skuFamily := keyvault.SkuFamilyA
+	skuName := keyvault.SkuNameStandard
+
+	//TODO: This value here is a random generated string to comply with `^[0-9a-fA-F]{8}(-[0-9a-fA-F]{4}){3}-[0-9a-fA-F]{12}$` regex.
+	// Tried random generator here which gets stored in the recordings and going to differ on each run resulting in test failure.
+	// We kinda need some static value here
+	str := "1C793267-c310-d4ae-7BD5-5Af5BEF875D3"
```

Comment:
You could add a `tc.Namer.RandomGUID()` function and use that instead of this, but it's not a big deal.

Reply:
I'll get this one replaced in the next PR.. making some changes on fetching the tenant id from envVar and hiding it in the recordings.